src/main/java/spoon/support/compiler/jdt/ParentExiter.java

```diff
@@ -445,10 +445,11 @@ public <T> void visitCtBinaryOperator(CtBinaryOperator<T> operator) {
 			} else if (jdtTreeBuilder.getContextBuilder().stack.peek().node instanceof StringLiteralConcatenation) {
 				CtBinaryOperator<?> op = operator.getFactory().Core().createBinaryOperator();
 				op.setKind(BinaryOperatorKind.PLUS);
-				op.setLeftHandOperand(operator.getRightHandOperand());
-				op.setRightHandOperand((CtExpression<?>) child);
+				op.setLeftHandOperand(operator.getLeftHandOperand());
+				op.setRightHandOperand(operator.getRightHandOperand());
 				op.setType((CtTypeReference) operator.getFactory().Type().STRING.clone());
-				operator.setRightHandOperand(op);
+				operator.setLeftHandOperand(op);
+				operator.setRightHandOperand(((CtExpression<?>) child));
```

Comment:
I have done exactly what I have explained [here](https://github.com/INRIA/spoon/pull/4809#issuecomment-1189865743). I ensured that the left operand takes the nested operator whenever possible.